### PR TITLE
Add Firefox completions

### PR DESCRIPTION
**Pull request by daanturo**
## Description

Add firefox completions

Fixes issue: None

## TODOs:

- [ ] Changes to fish usage are reflected in user documentation/manpages.
- [ ] Tests have been added for regressions fixed
- [x] User-visible changes noted in CHANGELOG.rst

### CODE REVIEW

File: share/completions/firefox.fish [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# X11 options
+complete -c firefox -l display -r -d "X display to use"
+complete -c firefox -l sync -d "Make X calls synchronous"
+complete -c firefox -l g-fatal-warnings -d "Make all warnings fatal"
+
+# Firefox options
+complete -c firefox -s h -l help -d "Print this message"
+complete -c firefox -s v -l version -d "Print Firefox version"
+complete -c firefox -l full-version -d "Print Firefox version, build and platform build ids"
+complete -c firefox -s P -r -d "Start with <profile>"
```
Comment (krobelus) [inline share/completions/firefox.fish:10]: Nice work, thanks! Some minor comments below:

I think we can even suggest possible profiles:

```suggestion
complete -c firefox -s P -r -d "Start this profile" -fa '(string replace -rf "Name=(.*)" \'$1\' < ~/.mozilla/firefox/profiles.ini)'
```

File: share/completions/firefox.fish [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+complete -c firefox -l migration -d "Start with migration wizard"
+complete -c firefox -l ProfileManager -d "Start with ProfileManager"
+complete -c firefox -l no-remote -d "Do not accept or send remote commands implies --new-instance"
+complete -c firefox -l new-instance -d "Open new instance, not a new window in running instance"
+complete -c firefox -l safe-mode -d "Disables extensions and themes for this session"
+complete -c firefox -l allow-downgrade -d "Allows downgrading a profile"
+complete -c firefox -l MOZ_LOG -r -d "Treated as MOZ_LOG=<modules> environment variable, overrides it"
+complete -c firefox -l MOZ_LOG_FILE -r -d "Treated as MOZ_LOG_FILE=<file> environment variable, overrides it"
+complete -c firefox -l headless -d "Run without a GUI"
+complete -c firefox -l jsdebugger -d "Open the Browser Toolbox. Defaults to the local build but can be overridden by a firefox path"
```
Comment (krobelus) [inline share/completions/firefox.fish:21]: There's a bug in firefox: --jsdebugger is listed twice. Maybe remove this one with the long description if the first sentence is enough for completions?

File: share/completions/firefox.fish [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+complete -c firefox -l search -r -d "Search <term> with your default search engine"
+complete -c firefox -l setDefaultBrowser -d "Set this app as the default browser"
+complete -c firefox -l first-startup -d "Run post-install actions before opening a new window"
+complete -c firefox -l kiosk -d "Start the browser in kiosk mode"
+complete -c firefox -l disable-pinch -d "Disable touch-screen and touch-pad pinch gestures"
+complete -c firefox -l jsconsole -d "Open the Browser Console"
+complete -c firefox -l jsdebugger -d "Open the Browser Toolbox"
+complete -c firefox -l devtools -d "Open DevTools on initial load"
+complete -c firefox -l marionette -d "Enable remote control server"
+complete -c firefox -l remote-debugging-port -d "Start the Firefox Remote Agent, which is a low-level remote debugging interface used for WebDriver BiDi and CDP"
```
Comment (krobelus) [inline share/completions/firefox.fish:40]: This description is quite long, how about just "Start the Firefox Remote Agent"?

I wonder if we should make the pager trim overly-long descriptions.

File: CHANGELOG.rst [hunk 4]
```diff
@@ -32,6 +32,9 @@ Completions
 
   - ``zig`` (:issue:`9083`)
 
+  - ``firefox``
+  - ``firefox-developer-edition``
```
Comment (krobelus) [inline CHANGELOG.rst:36]: please keep sorted alphabetically (before zig) and lose the linebreak

File: share/completions/firefox.fish [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+complete -c firefox -l jsdebugger -d "Open the Browser Toolbox"
+complete -c firefox -l wait-for-jsdebugger -d "Spin event loop until JS debugger connects"
+complete -c firefox -l start-debugger-server -d "Start the devtools server on a TCP port or Unix domain socket path"
+complete -c firefox -l browser -d "Open a browser window"
+complete -c firefox -l new-window -r -d "Open <url> in a new window"
+complete -c firefox -l new-tab -r -d "Open <url> in a new tab"
+complete -c firefox -l private-window -r -d "Open <url> in a new private window"
+complete -c firefox -l preferences -d "Open Preferences dialog"
+complete -c firefox -l screenshot -d "Save screenshot to <path> or in working directory"
+complete -c firefox -f -l window-size -d "Width and optionally height of screenshot"
```
Comment (krobelus) [inline share/completions/firefox.fish:30]: this should have `-r` (AKA `--require-parameter`)

File: share/completions/firefox.fish [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+complete -c firefox -l full-version -d "Print Firefox version, build and platform build ids"
+complete -c firefox -s P -r -d "Start this profile" -fa '(string replace -rf "Name=(.*)" \'$1\' < ~/.mozilla/firefox/profiles.ini)'
+complete -c firefox -l profile -r -d "Start with profile at <path>"
+complete -c firefox -l migration -d "Start with migration wizard"
+complete -c firefox -l ProfileManager -d "Start with ProfileManager"
+complete -c firefox -l no-remote -d "Do not accept or send remote commands implies --new-instance"
+complete -c firefox -f -l new-instance -d "Open new instance, not a new window in running instance"
+complete -c firefox -f -l safe-mode -d "Disables extensions and themes for this session"
+complete -c firefox -f -l allow-downgrade -d "Allows downgrading a profile"
+complete -c firefox -l MOZ_LOG -r -d "Treated as MOZ_LOG=<modules> environment variable, overrides it"
```
Comment (krobelus) [inline share/completions/firefox.fish:18]: small misunderstanding: I meant only the --MOZ_LOG should have `-f`. Not the other flags like `--new-instance`, they don't take any parameters anyway

I'll correct both and merge. Thanks for the contribution!

Comment (daanturo) [reply share/completions/firefox.fish]: Thank you.

**Pull request merged**